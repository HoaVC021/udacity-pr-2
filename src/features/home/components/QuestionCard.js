import React, { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';
import { Button, Card, CardContent, CardActions, Typography, Divider } from '@mui/material';

import { fDateTime } from '../../../utils/formatTime';

const QuestionCard = ({ question }) => {
  const navigate = useNavigate();

  const handleClick = useCallback(() => {
    navigate(`/questions/${question.id}`);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <Card variant='outlined'>
      <CardContent>
        <Typography textAlign='center' variant='h6'>{question.author}</Typography>
        <Typography textAlign='center' variant='body2' color='text.secondary' sx={{ my: 1 }}>
          {fDateTime(question.timestamp)}
        </Typography>
      </CardContent>
      <Divider />
      <CardActions>
        <Button onClick={handleClick} fullWidth variant='outlined' sx={{ mx: 1 }}>Show</Button>
      </CardActions>
    </Card>
  );
};

export default QuestionCard;
