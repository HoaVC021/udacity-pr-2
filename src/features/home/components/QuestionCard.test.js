import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import QuestionCard from './QuestionCard'; // Adjust the import path as necessary

// Mock the fDateTime utility
jest.mock('../../../utils/formatTime', () => ({
  fDateTime: jest.fn((timestamp) => new Date(timestamp).toLocaleString()),
}));

describe('QuestionCard Component', () => {
    const question = {
        id: '123',
        author: 'John Doe',
        timestamp: Date.now(),
    };

    it('renders question card correctly', () => {
        render(
            <MemoryRouter>
                <QuestionCard question={question} />
            </MemoryRouter>
        );

        // Check if the author's name is rendered
        expect(screen.getByText(/john doe/i)).toBeInTheDocument();

        // Check if the button is rendered
        expect(screen.getByRole('button', { name: /show/i })).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <QuestionCard question={question} />
            </MemoryRouter>
        );

        // Create a snapshot of the rendered component
        expect(asFragment()).toMatchSnapshot();
    });
});
