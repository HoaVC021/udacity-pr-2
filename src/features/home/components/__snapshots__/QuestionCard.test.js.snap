// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-91svcj-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1pot5ch-MuiTypography-root"
      >
        John Doe
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body2 css-179zhq7-MuiTypography-root"
      />
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-xm0xqa-MuiCardActions-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1jvkc28-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Show
      </button>
    </div>
  </div>
</DocumentFragment>
`;
